src/ng/filter/filter.js

```diff
@@ -44,9 +44,9 @@
  *
  *     The final result is an array of those elements that the predicate returned true for.
  *
- * @param {function(actual, expected)|true|false} [comparator] Comparator which is used in
- *     determining if the expected value (from the filter expression) and actual value (from
- *     the object in the array) should be considered a match.
+ * @param {function(actual, expected)|true|false} [comparator] A predicate to further refine the
+ *     search results of an object or string expression. It should be used to determine if the
+ *     actual value from the array is equal to the expected value from the filter expression.
```

Comment:
TBH, I feel this change is more confusing. E.g. `comparator` is **not** used to _"further refine the search results"_.  And _"should be used"_ sounds more like the user should do something (vs AngularJS using the function under the hood). And also `actual` is not necessarily a _"value from the array"_, but can be a property or sub-property of an item in the collection/array.